.github/workflows/lint.yaml

```diff
@@ -0,0 +1,27 @@
+name: Lint source code
+on:
+  - push
+  - workflow_dispatch
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Update no-OS submodule
+        run: |
+          git submodule update --init libraries/no-OS
+
+      - name: Run astyle on source code
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y astyle fd-find
+          set -e
+          formatted=$(fdfind -e c -e h . projects | xargs astyle --formatted --options=libraries/no-OS/ci/astyle_config)
```

Comment:
I think libraries needs to be libs as we have changed that as a folder name

Reply:
This is sitting on my branch from #16 that adds no-OS as a submodule under `libraries/`. Otherwise it would also require calling the script to clone the repos.

Reply:
got it. We have shorten the folder name length because we had faced some issues with make system where length of file path exceeds certain limits on the windows OS. That I believe was because of 'cmd' interface. Now as make build is using 'gitbash shell' cmd interface for windows, this issue might not be there. But for now, we can stick with folder name as 'libs'.
May be you need to rebase your branch with latest main branch of no-os repo?

Reply:
Haha, yikes, that sounds like a really annoying problem.

[I created the `libraries` directory in PCF](https://github.com/analogdevicesinc/precision-converters-firmware/pull/16/commits/b0d77fc208061cbf6d0732612b54bc34bb1e901c) to follow how no-OS does it (see https://github.com/analogdevicesinc/no-OS/tree/master/libraries).

Sorry if it is a bit confusing. I didn't want to add more changes to #16 since it already includes a lot so I created this pull request on top of that one.

Reply:
This new PR make sense but only thing is we may have to rebase all your branches with main branch before any merge in case of conflict. For this particular issue of folder name conflicts, let me see if we can change this folder name back to libraries so that you don't have to change in your files.

Reply:
No problem! I'll handle all the conflicts. We are also going to submit a new project today or tomorrow that I will need to update in my MR as well.